### PR TITLE
[DO NOT MERGE] Updates for HCP Packer docs

### PR DESCRIPTION
**Pull request by laurapacilio**
Add the introductory page content that includes HCP Packer | Add HCP Packer docs link to the navigation. This content is not yet ready to be merged.

### CODE REVIEW

File: website/content/docs/index.mdx [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
+
+## HCP Packer
+
+The HCP Packer registry bridges the gap between image factories and image deployments, allowing development and security teams to work together to create, manage, and consume golden images in a centralized way.
+
+The HCP Packer registry stores metadata about your images, including when they were created, where the image exists in the cloud, and what (if any) git commit is associated with your image build. You can use the registry to track information about the golden images your Packer builds produce, clearly designate which images are appropriate for test and production environments, and query for the right golden images to use in both Packer and Terraform configurations.
+
+HCP Packer is under active development, and we are currently offering a [public beta version](https://portal.cloud.hashicorp.com) to collect feedback and continue improving the product. We encourage you to try HCP Packer and submit your feedback.
+
+To get started, visit the [HCP Packer documentation](https://cloud.hashicorp.com/docs/packer) or try the [Get Started with HCP Packer](https://learn.hashicorp.com/collections/packer/packer/hcp-get-started) collection on HashiCorp Learn.
```
[the remark this answers is not in the document]
Comment (laurapacilio) [reply website/content/docs/index.mdx:21]: yes! these will be live when this goes up - thank you for flagging :-)

**Review by im2nguyen — APPROVED** [review]
Looks great!

File: website/content/docs/templates/hcl_templates/blocks/build/hcp_packer_registry.mdx [hunk 2]
```diff
@@ -1,17 +1,33 @@
...
+collection on HashiCorp Learn.
 
-The presence of a `hcp_packer_registry` block will enable the HCP Packer registry mode and all the builds within that build block
-will be pushed to the remote registry if the appropriate HCP credentials are set (`HCP_CLIENT_ID` and `HCP_CLIENT_SECRET`). If no HCP credentials are set Packer will fail the build and exit immediately to avoid any potential artifact drift between the build providers and the HCP Packer registry.
+
+The `hcp_packer_registry` block allows operators the ability to customize the
+metadata sent to HCP Packer Registry. It configures the details of an image
+that is created or updated within the HCP Packer registry.
+
+This block is availabel from version 1.7.7 of Packer.
```
Comment (nywilken) [inline website/content/docs/templates/hcl_templates/blocks/build/hcp_packer_registry.mdx:23]: Pending approval of 1.8.0 release. 
```suggestion
This block is available from version 1.8.0 of Packer.
```

File: website/content/docs/templates/hcl_templates/blocks/build/hcp_packer_registry.mdx [hunk 3]
```diff
@@ -1,17 +1,33 @@
...
+collection on HashiCorp Learn.
 
-The presence of a `hcp_packer_registry` block will enable the HCP Packer registry mode and all the builds within that build block
-will be pushed to the remote registry if the appropriate HCP credentials are set (`HCP_CLIENT_ID` and `HCP_CLIENT_SECRET`). If no HCP credentials are set Packer will fail the build and exit immediately to avoid any potential artifact drift between the build providers and the HCP Packer registry.
+
+The `hcp_packer_registry` block allows operators the ability to customize the
+metadata sent to HCP Packer Registry. It configures the details of an image
+that is created or updated within the HCP Packer registry.
+
+This block is available from version 1.8.0 of Packer.
```
Comment (SwampDragons) [inline website/content/docs/templates/hcl_templates/blocks/build/hcp_packer_registry.mdx:23]:
```suggestion
This block is available from version 1.7.7 of Packer.
```

**Pull request merged**